Mathlib/NumberTheory/Divisors.lean

```diff
@@ -514,3 +524,113 @@ theorem disjoint_divisors_filter_isPrimePow {a b : ℕ} (hab : a.Coprime b) :
   exact hn.ne_one (Nat.eq_one_of_dvd_coprimes hab han hbn)
 
 end Nat
+
+namespace Int
+variable {xy : ℤ × ℤ} {x y z : ℤ}
+
+-- Local notation for the embeddings `n ↦ n, n ↦ -n : ℕ → ℤ`
+local notation "natCast" => Nat.castEmbedding (R := ℤ)
+local notation "negNatCast" =>
+  Function.Embedding.trans Nat.castEmbedding (Equiv.toEmbedding (Equiv.neg ℤ))
+
+/-- Pairs of divisors of an integer as a finset.
+
+`z.divisorsAntidiag` is the finset of pairs `(a, b) : ℤ × ℤ` such that `a * b = z`.
+As a special case, `Int.divisorsAntidiag 0 = ∅`.
+
+O(|z|). Computed from `Nat.divisorsAntidiagonal`. -/
+def divisorsAntidiag : (z : ℤ) → Finset (ℤ × ℤ)
+  | (n : ℕ) =>
+    let s : Finset (ℕ × ℕ) := n.divisorsAntidiagonal
+    (s.map <| .prodMap natCast natCast).disjUnion (s.map <| .prodMap negNatCast negNatCast) <| by
+      simp +contextual [s, disjoint_left, eq_comm]
+  | negSucc n =>
+    let s : Finset (ℕ × ℕ) := (n + 1).divisorsAntidiagonal
+    (s.map <| .prodMap natCast negNatCast).disjUnion (s.map <| .prodMap negNatCast natCast) <| by
+      simp +contextual [s, disjoint_left, eq_comm, forall_swap (α := _ * _ = _)]
+
+@[simp]
+lemma mem_divisorsAntidiag :
+    ∀ {z} {xy : ℤ × ℤ}, xy ∈ divisorsAntidiag z ↔ xy.fst * xy.snd = z ∧ z ≠ 0
+  | (n : ℕ), ((x : ℕ), (y : ℕ)) => by
+    simp [divisorsAntidiag]
+    norm_cast
+    simp +contextual [eq_comm]
+  | (n : ℕ), (negSucc x, negSucc y) => by
+    simp [divisorsAntidiag, negSucc_eq, -neg_add_rev]
+    norm_cast
+    simp +contextual [eq_comm]
+  | (n : ℕ), ((x : ℕ), negSucc y) => by
+    simp [divisorsAntidiag, negSucc_eq, -neg_add_rev]
+    norm_cast
+    aesop
+  | (n : ℕ), (negSucc x, (y : ℕ)) => by
+    simp [divisorsAntidiag]
+    norm_cast
+    aesop
+  | .negSucc n, ((x : ℕ), (y : ℕ)) => by
+    simp [divisorsAntidiag]
+    norm_cast
+    aesop
+  | .negSucc n, (negSucc x, negSucc y) => by
+    simp [divisorsAntidiag, negSucc_eq, -neg_add_rev]
+    norm_cast
+    simp +contextual [eq_comm]
+  | .negSucc n, ((x : ℕ), negSucc y) => by
+    simp [divisorsAntidiag, negSucc_eq, -neg_add_rev]
+    norm_cast
+    aesop
+  | .negSucc n, (negSucc x, (y : ℕ)) => by
+    simp [divisorsAntidiag, negSucc_eq, -neg_add_rev]
+    norm_cast
+    simp +contextual [eq_comm]
+
+@[simp] lemma divisorsAntidiag_zero : divisorsAntidiag 0 = ∅ := rfl
+
+lemma prodMk_mem_divisorsAntidiag (hz : z ≠ 0) : (x, y) ∈ z.divisorsAntidiag ↔ x * y = z := by
+  simp [hz]
+
+lemma swap_mem_divisorsAntidiag : xy.swap ∈ z.divisorsAntidiag ↔ xy ∈ z.divisorsAntidiag := by
+  simp [mul_comm]
+
+/-- Simp normal form of `Nat.swap_mem_divisorsAntidiagonal`. -/
+@[simp]
+lemma swap_mem_divisorsAntidiagonal_aux :
```

Comment:
This isn't really an `aux` lemma. Can we just use prime? Also, can you please specify in the docstring that you mean the LHS is the simp normal form?

Reply:
I was following the nat version of the lemma

Reply:
I've prefixed `_simpNF` instead to show this isn't a lemma people should use by hand. Can you check you're happy with it. If so, feel free to merge